mvnxparser/mvnxStreamReader/mvnxStreamReader.cpp

```diff
@@ -0,0 +1,218 @@
+ *
+ */
+
+/**
+ * @file mvnxStreamReader.cpp
+ * @brief Validate and parse efficiently mvnx files
+ * @author Diego Ferigo
+ * @date 18/04/2017
+ */
+
+#include "mvnxStreamReader.h"
+
+using namespace std;
+using namespace xmlstream;
+using namespace xmlstream::mvnx;
+
+void mvnxStreamReader::printParsedDocument()
+{
+    // Initialize the XML file
+    QXmlStreamReader xml;
+    xmlFile.seek(0);
+    xml.setDevice(&xmlFile);
+
+    // Initialize the objects that will contain the metadata of the current
+    // element
+    QXmlStreamReader::TokenType tokenType;
+    string tokenString;
+    string elementName;
+    // string elementNS; // TODO: Namespace support
+    string elementText;
+    QXmlStreamAttributes elementAttributes;
+
+    while (!xml.atEnd()) {
+        xml.readNext();
+
+        tokenType          = xml.tokenType();
+        tokenString        = xml.tokenString().toStdString();
+        string elementName = xml.name().toString().toStdString();
+        // string ElementNS   = xml.namespaceUri().toString().toStdString();
+        string elementText = xml.text().toString().toStdString();
+        elementAttributes  = xml.attributes();
+
+        if (not xml.isCharacters()) {
+            cout << tokenString << ": " << elementName << endl;
+            if (not elementAttributes.empty()) {
+                for (auto element : elementAttributes)
+                    cout << "   " << element.name().toString().toStdString()
+                         << "=\"" << element.value().toString().toStdString()
+                         << "\"" << endl;
+            }
+        }
+
+        if (xml.isStartDocument() || xml.isEndDocument()) {
+            cout << endl;
+            continue;
+        }
+
+        // Print the content of the element
+        if (xml.isCharacters() && not xml.isWhitespace())
+            cout << tokenString << ": " << xml.text().toString().toStdString()
+                 << endl;
+    }
+}
+
+bool mvnxStreamReader::parse()
+{
+    // Initialize the XML file
+    QXmlStreamReader xml;
+    xmlFile.seek(0);
+    xml.setDevice(&xmlFile);
+
+    // Initialize the objects that will contain the metadata of the current
+    // element
+    QXmlStreamReader::TokenType tokenType;
+    string tokenString;
+    string elementName;
+    // string elementNS; // TODO: Namespace support
+    string elementText;
+    QXmlStreamAttributes elementAttributes;
+
+    // Sequentially parse the file
+    while (!xml.atEnd()) {
+        xml.readNext();
+
+        // Get the metadata of the element
+        tokenType   = xml.tokenType();
+        tokenString = xml.tokenString().toStdString();
+        elementName = xml.name().toString().toStdString();
+        // elementNS   = xml.namespaceUri().toString().toStdString();
+        elementText       = xml.text().toString().toStdString();
+        elementAttributes = xml.attributes();
+
+        // Handle the generated events
+        switch (tokenType) {
```

Comment:
Given the limited options we need for parsing MVNX files this is ok.
Probably if we have to code a more generic XML parsing library I would try to opt to a visitor pattern